read-fonts/src/tables/gvar.rs

```diff
@@ -57,9 +57,14 @@ impl<'a> GlyphVariationDataHeader<'a> {
 }
 
 impl<'a> Gvar<'a> {
-    pub fn data_for_gid(&self, gid: GlyphId) -> Result<FontData<'a>, ReadError> {
-        let range = self.data_range_for_gid(gid)?;
-        self.data.slice(range).ok_or(ReadError::OutOfBounds)
+    /// Return the raw data for this gid.
+    ///
+    /// If there is no variation data for the glyph, returns `Ok(None)`.
+    pub fn data_for_gid(&self, gid: GlyphId) -> Result<Option<FontData<'a>>, ReadError> {
+        match self.data_range_for_gid(gid)? {
```

Comment:
I believe `data_range_for_gid` should remain unchanged and this should return `Ok(None)` when the range is empty.

Reply:
And we should still return an error when the range is out of bounds.

Reply:
good call, I'd misunderstood the logic here.